app/controllers/application_controller.rb

```diff
@@ -1,4 +1,4 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
-  before_action :authenticate_user!
+  # before_action :authenticate_user!
```

Comment:
Tại sao lại bỏ authenticate ở đây.

Reply:
Mình không muốn user vào trang thì cần phải đăng nhập. Mình sẽ làm khi user muốn comment or add 1 quyển sách nào đó thì mình bắt đăng nhập. Còn không user sẽ có quyền xem trang bình thường

Reply:
Mình sẽ làm cần đăng nhập khi người dùng có những action như là like, add to cart , comment , rating ....

Reply:
có nên để tính năng đăng nhập nhanh bằng facebook hoặc gmail không. để tiện thông báo cho người dùng